### ISSUE TITLE
Symfony 4 - Encrypted and decrypted key is the same.

### ISSUE THREAD

**Issue by Volmarg** [mogilvie/EncryptBundle#7]
Hello,

I've just added Your bundle to my symfony project 4.3 i think. I will skip issues with key generating but I've managed to make 256 bit one.

Now I've got a problem that actually encrypted and decrypted key is the same.
Am i doing something wrong or any idea?

![image](https://user-images.githubusercontent.com/37488679/59703276-d355a380-91f9-11e9-9007-9116f217bfbb.png)
![image](https://user-images.githubusercontent.com/37488679/59703289-da7cb180-91f9-11e9-9292-69f685cd9064.png)

I'm just doing tests now, so You can recreate it with this key:
`encrypt_key: 'kXp2s5v8y/A?D(G+KbPehVmYq3t6whVmYq3tt6wShVw9z'`
**encrypted string:** test

Same for twig:
`{{ 'WlCj153fQK1yzp1qZXByLBaUL9T2vw7nq7fbSXruZ8Q=' | decrypt }}`
**Result**: `WlCj153fQK1yzp1qZXByLBaUL9T2vw7nq7fbSXruZ8Q=`

**Comment by Volmarg** [comment]
Yeah it's a bug or some incompatibillity.
![image](https://user-images.githubusercontent.com/37488679/59704372-3fd1a200-91fc-11e9-9f29-10a2f290a9ca.png)

This saved such string into database:
`oQzacpgt4sPRTyu0kHMOYf7tc1fh1YQecmZ+tO4IJfM=<ENC>`

Now when i use this string for tests
````
 $decrypted = $this->encryptor->decrypt('oQzacpgt4sPRTyu0kHMOYf7tc1fh1YQecmZ+tO4IJfM=<ENC>');
````

![image](https://user-images.githubusercontent.com/37488679/59704441-6b548c80-91fc-11e9-887f-df698d50cd61.png)


So it looks like the encrypted keys in controller are missing this `<ENC>` thing,

**Comment by mogilvie** [comment]
Hi Volmarg,
There is only one encryption key, unique to your installation. The key is used to both encrypt and decrypt the stored value, see [the encryptor](https://github.com/mogilvie/EncryptBundle/blob/master/Encryptors/OpenSslEncryptor.php)

When decrypting the decryptor looks for 5 characters at the end of the persisted string "\<ENC\>".  If it finds this string fragment, then it knows the value is encrypted and runs the decrypt process. Otherwise it returns the raw stored value

         // If the value does not have the suffix <ENC> then ignore.
        if(substr($data, -5) !== DoctrineEncryptSubscriberInterface::ENCRYPTED_SUFFIX) {
            return $data;
        }

Your initial test didn't contain the suffix  "\<ENC\>" so it wasn't decrypted.

**Comment by Volmarg** [comment]
Thanks for response.

Yeah I've figured it out but what I mean is that encryptor actually created that string without ENC. And then decryptor couldnt handle it.

**Comment by mogilvie** [comment]
OK. Understand. I'll look into it tonight.

**Comment by mogilvie** [comment]
Just had a quick look now.
The \<ENC\> suffix is added to the the encrypted value during the doctrine persistence event.
As you are not persisting the encypted string in the tests, then it is not being appended.
I think it would make more sense to add the suffix in the decryptor/encryptor.
The encryptor needs to add the suffix, and to maintain BC the DoctrineEncryptSubscriber will need a check to add the suffix only if it doesnt already exist.
I can make the changes on friday, or if you need it done quickly, then submit a pull request.
Thanks for raising.

**Comment by Volmarg** [comment]
Nah it does the job for me with entity. Just had this issue so I wrote info. Thanks.

**Comment by mogilvie** [comment]
Fixed in v1.1.1
